### PR TITLE
feat(ui-ux): add ui changes to voting flow

### PR DESCRIPTION
**Pull request by veralygit**
#### What this PR does / why we need it:
1) update proposal details page voting flow progress bar to tug of war progress bar 
2) included approval threshold indicator in progress bar
3) updated positioning of labels in voting flow 

#### Which issue(s) does this PR fixes?:

Fixes #

#### Sample Links & Screenshots:

Link:

<details>
<summary>Desktop Screenshot</summary>

</details>
<details>
<summary>Mobile Screenshot</summary>

</details>

#### Developer Checklist:

- [x] Read your code changes at least once
- [x] Tested on multiple web browsers
- [x] Tested responsiveness (e.g, iPhone, iPad, Desktop)
- [x] No console errors
- [ ] Unit tests\*
- [ ] Added e2e tests\*

### CODE REVIEW

File: src/pages/governance/_components/VotingResult.tsx [hunk 1]
```diff
@@ -91,91 +91,77 @@ export function VotingResult({
...
                 <Progress
-                  value={percNo.toNumber()}
+                  yesValue={percYes.toNumber()}
+                  noValue={percNo.toNumber()}
+                  approvalThreshold={Number(
+                    proposal.approvalThreshold.replace("%", "")
+                  )}
                   containerClass="bg-gray-100 dark:bg-dark-gray-200"
-                  contentClass="bg-gray-300 dark:bg-dark-gray-400"
+                  // contentClass="bg-gray-300 dark:bg-dark-gray-400"
```
Comment (kyleleow) [inline src/pages/governance/_components/VotingResult.tsx:110]: we need this?

File: src/pages/governance/_components/VotingResult.tsx [hunk 2]
```diff
@@ -386,44 +341,63 @@ export function VotingResult({
...
-    <div
-      className={classNames(
-        "h-3 rounded-[54px] w-full bg-gray-200 dark:bg-dark-gray-200 relative overflow-hidden",
-        containerClass
-      )}
-    >
+    <div className="h-5 relative">
+      <div
+        style={{ width: `${approvalThreshold}%` }}
+        className="absolute left-0 -top-1 h-5 border-r border-gray-200 dark:border-dark-gray-200"
```
Comment (kyleleow) [inline src/pages/governance/_components/VotingResult.tsx:358]: check height again

File: src/pages/governance/_components/VotingResult.tsx (hunk 1 above)
Comment (veralygit) [reply src/pages/governance/_components/VotingResult.tsx:110]: ahh i caught this just now going to push

File: src/pages/governance/_components/VotingResult.tsx [hunk 3]
```diff
@@ -90,92 +90,83 @@ export function VotingResult({
...
-                      "grow text-gray-600 dark:text-dark-gray-600 text-sm"
-                    )}
-                  >
-                    {percNo.toFixed(2)}%
-                  </span>
+                <div className="flex flex-row">
+                  <div className="flex flex-col grow">
+                    <span
+                      className={classNames(
+                        "text-sm",
```
Comment (kyleleow) [inline src/pages/governance/_components/VotingResult.tsx:115]: check font size

File: src/pages/governance/_components/VotingResult.tsx [hunk 4]
```diff
@@ -90,92 +90,83 @@ export function VotingResult({
...
+                      thousandSeparator=","
+                      displayType="text"
+                      suffix=" votes"
+                      className="text-sm text-gray-600 dark:text-dark-gray-600"
+                    />
+                  </div>
+                  <div className="flex flex-col">
+                    <span
+                      className={classNames(
+                        "grow  text-sm text-right",
```
Comment (kyleleow) [inline src/pages/governance/_components/VotingResult.tsx:135]:
```suggestion
                        "grow text-sm text-right",
```

File: src/pages/governance/_components/VotingResult.tsx [hunk 5]
```diff
@@ -90,92 +90,83 @@ export function VotingResult({
...
-                    displayType="text"
-                    className="text-sm text-gray-600 dark:text-dark-gray-600"
-                  />
+                    <NumericFormat
+                      value={voteCounts.no}
+                      fixedDecimalScale
+                      thousandSeparator=","
+                      displayType="text"
+                      suffix=" votes"
+                      className="text-sm text-gray-600 dark:text-dark-gray-600"
```
Comment (kyleleow) [inline src/pages/governance/_components/VotingResult.tsx:150]: check font size

File: src/pages/governance/_components/VotingResult.tsx [hunk 6]
```diff
@@ -90,92 +90,83 @@ export function VotingResult({
...
-              <div className="flex flex-col gap-y-2 w-full">
-                <span className="md:text-lg text-sm font-semibold text-gray-900 dark:text-dark-gray-900">
-                  Yes
-                </span>
-                <Progress
-                  value={percYes.toNumber()}
-                  containerClass="bg-gray-100 dark:bg-dark-gray-200"
-                  contentClass="bg-gray-300 dark:bg-dark-gray-400"
-                />
+              <div className="flex flex-col gap-y-1 w-full">
```
Comment (kyleleow) [inline src/pages/governance/_components/VotingResult.tsx:93]: - let's not use `gap-*` when the spacing between the elements are not even
- also check the vertical spacing of all the child element in this container

File: src/pages/governance/_components/VotingResult.tsx [hunk 7]
```diff
@@ -471,16 +450,34 @@ function LabelWithInfoTooltipAndChecks({
...
-              : "fill-gray-300 dark:fill-dark-gray-400"
-          )}
-        />
+        {proposalStatus === GovernanceProposalStatus.VOTING ? (
+          <CircularCheckIcon
+            width={14}
+            height={14}
+            className={classNames(
+              "",
+              baseValue!.isGreaterThanOrEqualTo(BigNumber(comparatorValue))
```
Comment (kyleleow) [inline src/pages/governance/_components/VotingResult.tsx:459]:
```suggestion
              baseValue!.isGreaterThanOrEqualTo(comparatorValue)
```

File: src/pages/governance/_components/VotingResult.tsx [hunk 8]
```diff
@@ -471,16 +450,34 @@ function LabelWithInfoTooltipAndChecks({
...
+                ? "fill-green-600 dark:fill-dark-green-500"
+                : "fill-gray-300 dark:fill-dark-gray-400"
+            )}
+          />
+        ) : (
+          <>
+            {baseValue!.isGreaterThanOrEqualTo(BigNumber(comparatorValue)) ? (
```
Comment (kyleleow) [inline src/pages/governance/_components/VotingResult.tsx:466]:
```suggestion
            {baseValue!.isGreaterThanOrEqualTo(comparatorValue) ? (
```

File: src/pages/governance/_components/VotingResult.tsx [hunk 9]
```diff
@@ -386,56 +346,75 @@ export function VotingResult({
...
+  proposalStatus,
+  suffix,
   toolTipDesc,
   decimalPlace,
 }: {
   labelTitle: string;
   value: BigNumber;
-  suffix?: string;
+  baseValue?: BigNumber;
   comparatorValue?: BigNumber;
```
Comment (kyleleow) [inline src/pages/governance/_components/VotingResult.tsx:403]: - from what i see, comparatorValue is the same as value, can we reuse it and dont have to reintroduce another repeated variable?
- to me `baseValue` is more like the `comparatorValue`, assuming `value` is the one we are going to display

File: src/pages/governance/_components/VotingResult.tsx [hunk 10]
```diff
@@ -471,16 +450,34 @@ function LabelWithInfoTooltipAndChecks({
...
-              ? "fill-green-600 dark:fill-dark-green-500"
-              : "fill-gray-300 dark:fill-dark-gray-400"
-          )}
-        />
+        {proposalStatus === GovernanceProposalStatus.VOTING ? (
+          <CircularCheckIcon
+            width={14}
+            height={14}
+            className={classNames(
+              "",
```
Comment (kyleleow) [inline src/pages/governance/_components/VotingResult.tsx:458]: we need this?

File: src/pages/governance/_components/VotingResult.tsx [hunk 11]
```diff
@@ -192,65 +183,34 @@ export function VotingResult({
...
 
-              <div className="flex flex-col">
+              <div className="flex flex-col gap-y-5">
                 <LabelWithInfoTooltipAndChecks
-                  labelTitle="Neutral votes"
-                  value={BigNumber(voteCounts.neutral)}
-                  // toolTipDesc="Included as part of the total votes submitted, excluded from min. approval." // TODO: uncomment when blockchain fixes neutral votes as no bug
+                  labelTitle="Min. required approval rate"
+                  value={
+                    new BigNumber(proposal.approvalThreshold.replace("%", ""))
```
Comment (kyleleow) [inline src/pages/governance/_components/VotingResult.tsx:193]: i see we have `new BigNumber(..` and sometimes `BigNumber(..` in this code, let's be consistent

File: src/pages/governance/_components/VotingResult.tsx [hunk 12]
```diff
@@ -90,92 +90,82 @@ export function VotingResult({
...
-                <div className="flex flex-row">
-                  <span
-                    className={classNames(
-                      "grow text-gray-600 dark:text-dark-gray-600 text-sm"
-                    )}
-                  >
-                    {percYes.toFixed(2)}%
+              <div className="flex flex-col w-full">
+                <div className="flex flex-row mb-1">
+                  <span className="md:text-base text-sm font-semibold text-gray-900 dark:text-dark-gray-900 grow">
```
Comment (kyleleow) [inline src/pages/governance/_components/VotingResult.tsx:95]: check the spacing with the progress bar

File: src/pages/governance/_components/VotingResult.tsx [hunk 13]
```diff
@@ -90,92 +90,82 @@ export function VotingResult({
                 "mb-6 lg:pb-[22px] lg:pr-0 md:pr-8 md:pb-0 pb-[22px] lg:w-full w-full md:w-1/2 lg:border-b-[0.5px] md:border-b-0 border-b-[0.5px] dark:border-dark-gray-300"
```
Comment (kyleleow) [inline src/pages/governance/_components/VotingResult.tsx:90]:
```suggestion
                "mb-[22px] lg:pb-[22px] lg:pr-0 md:pr-8 md:pb-0 pb-[22px] lg:w-full w-full md:w-1/2 lg:border-b-[0.5px] md:border-b-0 border-b-[0.5px] dark:border-dark-gray-300"
```

File: src/pages/governance/_components/VotingResult.tsx (hunk 6 above)
Comment (veralygit) [reply src/pages/governance/_components/VotingResult.tsx]: checked that the padding between the bar and the labels are 10px evenly updated to use gap-y-[10px] removed the extra spacing for the threshold line

File: src/pages/governance/_components/VotingResult.tsx (hunk 7 above)
Comment (veralygit) [reply src/pages/governance/_components/VotingResult.tsx]: removed all BigNumber to use new BigNumber

File: src/pages/governance/_components/VotingResult.tsx (hunk 8 above)
Comment (veralygit) [reply src/pages/governance/_components/VotingResult.tsx]: updated to remove BigNumber

File: src/pages/governance/_components/VotingResult.tsx (hunk 9 above)
Comment (veralygit) [reply src/pages/governance/_components/VotingResult.tsx:399]: updated variables for component

File: src/pages/governance/_components/VotingResult.tsx (hunk 10 above)
Comment (veralygit) [reply src/pages/governance/_components/VotingResult.tsx]: removed

File: src/pages/governance/_components/VotingResult.tsx (hunk 11 above)
Comment (veralygit) [reply src/pages/governance/_components/VotingResult.tsx:192]: removed all BigNumber to use new BigNumber

File: src/pages/governance/_components/VotingResult.tsx (hunk 12 above)
Comment (veralygit) [reply src/pages/governance/_components/VotingResult.tsx:92]: checked that the padding between the bar and the labels are 10px evenly updated to use gap-y-[10px] removed the extra spacing for the threshold line

File: src/pages/governance/_components/VotingResult.tsx (hunk 13 above)
Comment (veralygit) [reply src/pages/governance/_components/VotingResult.tsx]: updated

File: src/pages/governance/_components/VotingResult.tsx [hunk 14]
```diff
@@ -452,35 +425,55 @@ function LabelWithInfoTooltipAndChecks({
...
+              {comparatorValue!.isGreaterThanOrEqualTo(value) ? (
+                <CircularCheckIcon
+                  width={14}
+                  height={14}
+                  className={classNames(
+                    "fill-green-600 dark:fill-dark-green-500"
+                  )}
+                />
+              ) : (
+                <>
```
Comment (kyleleow) [inline src/pages/governance/_components/VotingResult.tsx:470]: we need this?

File: src/pages/governance/_components/VotingResult.tsx [hunk 15]
```diff
@@ -452,35 +425,55 @@ function LabelWithInfoTooltipAndChecks({
...
-              ? "fill-green-600 dark:fill-dark-green-500"
-              : "fill-gray-300 dark:fill-dark-gray-400"
-          )}
+      <div className="flex items-center">
+        <NumericFormat
+          value={value.toFixed(decimalPlace)}
+          fixedDecimalScale
+          thousandSeparator=","
+          displayType="text"
+          suffix={suffix ?? ""}
```
Comment (kyleleow) [inline src/pages/governance/_components/VotingResult.tsx:444]: can we use this?
```suggestion
          suffix={suffix}
```

**Pull request merged**